src/include/miscadmin.h

```diff
@@ -615,6 +616,13 @@ extern void GpRecoveryFromError(void);
 	Assert(dispatch_nest_level >  0); \
 	dispatch_nest_level--; \
 } while(0)
+#define SAVE_AND_ENSURE_DISPATCH()	do { \
+	saved_dispatch_nest_level = dispatch_nest_level;	\
+	dispatch_nest_level = 0;	\
+} while(0)
+#define RESTORE_DISPATCH()	do { \
+	dispatch_nest_level = saved_dispatch_nest_level; \
+} while(0)
```

Comment:
Could you add a usage on how to use the two macros in  the code? It looks like we could make value as an argument rather than a global variable.